src/mhd2xds.coffee

```diff
@@ -22,6 +39,13 @@ exports.mhd1Metadata2Xds = (metadata) ->
     seconds = '0' + seconds
   now = "#{year}#{month}#{day}#{hours}#{minutes}#{seconds}"
 
+  if metadata.documentEntry.patientId.indexOf '&' is -1 and metadata.documentEntry.patientId.indexOf 'ZAF' isnt -1
+    sourceId = metadata.documentEntry.patientId.replace 'ZAF', 'ZAF&0.0.0&ISO'
```

Comment:
actually come to think of it - maybe you should use the `root` in the CDA id as the assigning auth and just use the extension as is for the identifier: `{escapedExt}^^^&{root}&ISO` where `escapedExt = extension.replace('^', '\S\')` (i.e. [hl7 escaped](http://www.hl7standards.com/blog/2006/11/02/hl7-escape-sequences/))